source/base/bounding_box.cc

```diff
@@ -55,42 +52,53 @@ BoundingBox<spacedim, Number>::merge_with(
                  other_bbox.boundary_points.first[i]);
       this->boundary_points.second[i] =
         std::max(this->boundary_points.second[i],
-                 other_bbox.boundary_points.second[i]);
+                    other_bbox.boundary_points.second[i]);
     }
 }
 
-
+template <int spacedim, typename Number>
+bool
+BoundingBox<spacedim, Number>::is_neighbor(
+  const BoundingBox<spacedim, Number> &other_bbox,
+  const double                         tolerance) const
+{
+  for (unsigned int i = 0; i < spacedim; ++i)
+    {
+      // testing if the boxes are close enough to intersect
+      if ((other_bbox.boundary_points.second[i] <
+           this->boundary_points.first[i] - tolerance * side_length(i)) ||
+          (other_bbox.boundary_points.first[i] >
+           this->boundary_points.second[i] + tolerance * side_length(i)))
+        return false;
+    }
+  return true;
+}
```

Comment:
This additional function is ok.

Reply:
I would like to suggest a different name for this new function, e.g., `have_nonzero_intersection` or `intersection_is_nonzero` better describes what this function does.

Reply:
If we change the name, I think we would also have to change the name of the corresponding ```NeighborType``` to stay consistent, which might break third party code. 
That said, I would prefer the proposed name as well.